docs/database-classes.md

```diff
@@ -1,348 +1,283 @@
-
-The database schema in Starcounter is defined by C\# classes with the `[Database]` attribute:
+We can define the schema of a Starcounter database table in an application by defining **public abstract C# classes** decorated with the `DatabaseAttribute` attribute. We call these types **database classes**. The names and namespaces we give to these C# classes will be the names and namespaces of the corresponding database tables.
 
 ```csharp
+using System;
 using Starcounter.Database;
 
 [Database]
 public abstract class Person
 {
-    public abstract string FirstName { get; set; }
-    public abstract string LastName { get; set; }
-    public string FullName => $"{FirstName} {LastName}";
+    public abstract string Name { get; set; }
 }
 ```
 
-All instances of database classes created with the `IDatabaseContext.Insert` method are stored persistently.
+All instances of these database classes created with the `IDatabaseContext.Insert<T>()` method are stored in the Starcounter database persistently.
 
 ## Constructors
 
-Database classes support default constructors. Non-private default constructors are called when a new instance is created with `IDatabaseContext.Insert`.
+Database classes support default constructors, which is useful for adding logic that is run each time an instance of the database class is created. Public default constructors of database classes are called whenever a new instance of the class is created with `IDatabaseContext.Insert<T>()`.
 
-For example, this is a valid database class with a constructor:
+For exaαmple, here is our `Person` class with a custom default constructor:
 
 ```csharp
+using System;
 using Starcounter.Database;
 
 [Database]
-public abstract class Order
+public abstract class Person
 {
-    public Order()
+    public abstract string Name { get; set; }
+    public abstract DateTime CreatedAt { get; set; }
+
+    public Person()
     {
-        this.Created = DateTime.Now;
+        CreatedAt = DateTime.UtcNow;
     }
-
-    public abstract DateTime Created { get; set; }
 }
 ```
 
-It's possible to have constructors with parameters, although, they are never called when using `IDatabaseContext.Insert`. Constructors with parameters in database classes can be useful for unit testing purposes when you want to inject dependencies or other arguments into a class. If you add a constructor with parameters to a database class, you also have to add a default constructor.
-
-**Warning**: All C\# access modifiers are accepted for constructors, except for `internal`. Using it will throw `ScErrSchemaCodeMismatch (SCERR4177)` exception.
+It's allowed to have non-default constructors in database class definitions as well, although they are not called by `IDatabaseContext.Insert<T>()`. Instead they can be useful in unit testing scenarios when you want to inject dependencies or other arguments into a class instance. If you add a constructor with parameters to a database class, you also have to add a default constructor.
```

Comment:
I am a bit confused here. What is the point of having such constructors in abstract classes? Would not it make more sense to have the custom constructors in the derived non database classes, defined for the sake of unit testing?

Reply:
I guess it was a relic from previous versions of the docs. I'll remove that section

Reply:
Done